backend/PythonClient/server/simulation_server.py

```diff
@@ -33,90 +36,53 @@
-            all_monitors_pass = True
-            for path in paths_to_check:
-                flytopoints_path = os.path.join(path, 'FlyToPoints')
-                if os.path.exists(flytopoints_path) and os.path.isdir(flytopoints_path):
-                    for monitor in monitors:
-                        if not check_monitor_pass_fail(flytopoints_path, monitor):
-                            all_monitors_pass = False
-                            break
-                    if not all_monitors_pass:
-                        break
-                global_monitors_path = os.path.join(path, 'GlobalMonitors')
-                if os.path.exists(global_monitors_path) and os.path.isdir(global_monitors_path):
-                    for monitor in global_monitors:
-                        if not check_monitor_pass_fail(global_monitors_path, monitor):
-                            all_monitors_pass = False
-                            break
-                    if not all_monitors_pass:
-                        break
-            if all_monitors_pass:
-                pass_count = drone_count
-            else:
-                fail_count = drone_count
-            report_files.append({
-                'filename': file,
-                'contains_fuzzy': contains_fuzzy,
-                'drone_count': drone_count,
-                'pass': pass_count,
-                'fail': fail_count
-            })
-        else:
+
+    for blob in blobs:
+        if blob.name.endswith('.txt'):
+            file_name = blob.name.split('/')[-1]
+            drone_count = count_drone_files(bucket, file_name)
+            all_monitors_pass = check_monitor_pass_fail(bucket, file_name, monitors)
             report_files.append({
-                'filename': file,
+                'filename': file_name,
                 'contains_fuzzy': False,
-                'drone_count': 0,
-                'pass': 0,
-                'fail': 0
+                'drone_count': drone_count,
+                'pass': drone_count if all_monitors_pass else 0,
+                'fail': 0 if all_monitors_pass else drone_count
             })
+    
+    # # Reports file
+    # reports_path = os.path.join(os.path.expanduser("~"), "Documents", "AirSim", "report")
+    # if not os.path.exists(reports_path) or not os.path.isdir(reports_path):
+    #     return 'Reports directory not found', 404
+    def check_monitor_pass_fail(bucket, file_name, monitors):
+        monitor_pass = True
+        blob = bucket.blob(f'reports/{file_name}')
+        content = blob.download_as_text()
+
+        for monitor in monitors:
+            if monitor in content:
+                if 'FAIL' in content:
+                    monitor_pass = False
+                    break
+        return monitor_pass
+        
+    def count_drone_files(bucket, report_file_name):
+        count = 0
+        flytopoints_prefix = f'reports/{report_file_name}/FlyToPoints/'
+        blobs = bucket.list_blobs(prefix=flytopoints_prefix)
```

Comment:
These functions (`check_monitor_pass_fail` and `count_drone_files`) are still using local file operations. These need to be reimplemented to work with GCS's API by fetching and parsing blob data instead of using local paths.